docs/all_development/contributing/index.rst

```diff
@@ -0,0 +1,48 @@
+###############################################
+Contributing to Open edX code and documentation
+###############################################
+
+Overview
+========
```

Comment:
Note: I think I was just used to seeing the numbers in the legacy docs, but they shouldn’t be required if we fix the theme.

Reply:
I don't like the numbering: I think it creates visual noise, which is distracting. 

I'm not happy with the way I've done the outline levels. I'd like to make it consistent as follows: the top level toctrees should have 4 possible levels, mid-level TOC trees perhaps 2, and bottom-level 1. 

I'm also wondering if it's necessary to specify the maximum number of levels on the lower level index.rsts: in other words, let the top level max of 4 be the controlling factor.

Reply:
I've tried adding four heading levels to a topic with different maxdepth settings, both in the root index.rst and in the index.rst for the directory in which the stub appears.

The result is always the same: the headings in the stub do not appear in the TOC. 

This surprised me because I've seen it work elsewhere: the "contributing" index.rst.

However, of you look at contributing/index.rst, it does NOT follow the cheatsheet format, but it renders more predictably in the TOC. 

This looks to me like it comes from a documentation bug, which also appears here: https://thomas-cokelaer.info/tutorials/sphinx/rest_syntax.html#headings

I think it's best that we update our own cheatsheet documentation to indicate what actually yields expected results.

Reply:
I deleted a comment because it contained incorrect information from my testing.

No matter how or where a maxdepth is specified, or whether you followed the heading style in the cheatsheet, headings in topics at level 3 do not show up in the TOC. 

See Information for All Open EdX Developers => Technology common to core and extensions => Topic with subheadings to test TOC display. Compare with Information for All Open EdX Developers => Contributing to Open edX code and documentation. I'm about to push it now so you can see it.

Reply:
See commit 6ba1b49 for the test of the TOC display.

Reply:
I think you only will want depth of 2 in the TOCs, but you can play with it.
What I found is missing, is that you'll want to add the following line:
```
html_theme_options['navigation_depth'] = 4
``` 
Add the line here: https://github.com/edx/edx-developer-docs/blob/39529c105e5deee112bf768884bf90149df7ee18/docs/conf.py#L109

Reply:
> I think you only will want depth of 2 in the TOCs, but you can play with it.
> What I found is missing, is that you'll want to add the following line:
> 
> ```
> html_theme_options['navigation_depth'] = 4
> ```
> 
> Add the line here:
> 
> https://github.com/edx/edx-developer-docs/blob/39529c105e5deee112bf768884bf90149df7ee18/docs/conf.py#L109

I added the line above, and now there's something ugly that happens in the TOC when you click the equivalent of the index.rst that contains a link to the topic with headings.

I can't find the Open edX cheatsheet anywhere, not even in my browser history, so I have specified headings per this document: https://thomas-cokelaer.info/tutorials/sphinx/rest_syntax.html#headings

I have tried addressing the UI bug with all kinds of manipulations on max-length and I can't get it to go away. 

I'm not going to push the changes: we can take a look at it tomorrow or when you have time.

Reply:
> I think you only will want depth of 2 in the TOCs, but you can play with it.
> What I found is missing, is that you'll want to add the following line:
> 
> ```
> html_theme_options['navigation_depth'] = 4
> ```
> 
> Add the line here:
> 
> https://github.com/edx/edx-developer-docs/blob/39529c105e5deee112bf768884bf90149df7ee18/docs/conf.py#L109

I don't like what happens by default in the TOC when you click the index.rst that has a link to a topic with headings. I don't think GitHub will allow me to attach screenshots, unfortunately. 

I'm thinking that a global max depth of 2 makes the most sense for the TOC. 

I assume then that all the references to max-depth in the individual index.rst files should be removed?

Reply:
Rob and I discussed .... setting navigation-depth to 4 so that UI (theme) problem is obvious when present.

Reply:
> Rob and I discussed .... setting navigation-depth to 4 so that UI (theme) problem is obvious when present.

See commit b2689cc